app/models/space.rb

```diff
@@ -126,6 +128,20 @@ def copy(user)
     return space
   end
 
+
+  def needs_checkpoint?
+    checkpoint = checkpoints.order("created_at").last
+    return false unless checkpoint && graph
+    15.minutes.ago >= checkpoint.created_at
+  end
+
+  def take_checkpoint
+    checkpoints.create name: name, description: description, graph: graph
+    if checkpoints.count > 3000 # 3000 at 15 minute intervals in 1 month
```

Comment:
What?  I doubt they will be continuously editing for 1 month.  For now, how about 30?  Can scale after